src/components/number-input/number-input.tsx

```diff
@@ -0,0 +1,253 @@
+
+        this.commit(next);
+    }
+
+    private handleFocus: React.FocusEventHandler<HTMLElement> =
+        e => this.setState({focus: true});
+
+    private handleBlur: React.FocusEventHandler<HTMLElement> =
+        e => this.setState({focus: false});
+
+    private handleIncrement: React.MouseEventHandler<HTMLElement> =
+        e => this.stepValue(e, Direction.Increase);
+
+    private handleDecrement: React.MouseEventHandler<HTMLElement> =
+        e => this.stepValue(e, Direction.Decrease);
+
+    private handleInputKeyDown: React.KeyboardEventHandler<HTMLElement> =
+        e => {
+            switch (e.keyCode) {
+                case KeyCodes.UP:
+                    this.stepValue(e, Direction.Increase);
+                    e.preventDefault();
+                    break;
+                case KeyCodes.DOWN:
+                    this.stepValue(e, Direction.Decrease);
+                    e.preventDefault();
+                    break;
+                case KeyCodes.ENTER:
+                    this.commit(this.state.value);
+                    e.preventDefault();
+                    break;
+                case KeyCodes.ESCAPE:
+                    this.revert();
+                    e.preventDefault();
+                    break;
+            }
+        }
+
+    private handleInputBlur: React.FocusEventHandler<HTMLInputElement> =
+        e => this.commit(this.state.value);
+
+    private handleInputChange: React.ChangeEventHandler<HTMLInputElement> =
+        e => {
+            const {onInput} = this.props;
+            const value = e.target.value;
+            const next = value !== '' ?
+                Number(e.target.value) :
+                undefined;
+
+            this.updateValue(next);
+            onInput!(value);
+        }
+
+    componentWillReceiveProps({value}: NumberInputProps) {
+        const {min, max} = this.props;
+
+        if (value !== this.state.value) {
+            this.committed = true;
+            this.setState({value});
+            if (isNumber(value) && (value as number < min! || value > max!)) {
+                console.error('value is out of range');
+            }
+        }
+    }
+
+    render() {
+        const {value, focus, error} = this.state;
+        const {step, min, max, onInput, children, ...inputProps} = this.props;
+        const disableIncrement = inputProps.disabled || (isNumber(value) && value >= max!);
+        const disableDecrement = inputProps.disabled || (isNumber(value) && value <= min!);
+        const {prefix, suffix} = getAffix(children);
+
+        return <div
+            cssStates={{
+                disabled: Boolean(this.props.disabled),
+                focus, error
+            }}
+            onFocus={this.handleFocus}
+            onBlur={this.handleBlur}
+        >
+            {prefix.length > 0 ?
+                <div className="prefix">
+                    {prefix}
+                </div> : null
+            }
```

Comment:
I don't see <Affix /> holding anything useful, though. I suggest we extract it once have a clearer picture on how we slot children.